src/layers/domain/core/product_id.py

```diff
@@ -0,0 +1,48 @@
+import random
+import re
+
+"""
+The Product Id is a human readable identifier presented in the form of 3 x 3
+letter alphanumeric words. e.g. XXX-XXX-XXX.
+
+In order to reduce human error:
+1. All values are uppercase
+2. similar characters have been removed from the set of available characters.
+  e.g. 0/O/Q, 1/I, S/5, B/8, and Z/2
+
+This leaves 26+10-11 = 25 unique characters
+9 characters per id gives 3.8 billion unique values: 25^9 = 3,814,697,265,625
+"""
+
+PRODUCT_ID_CHARS = "ACDEFGHJKLMNPRTUVWXY34679"
+PRODUCT_ID_REGEX = (
+    f"^[{PRODUCT_ID_CHARS}]{{3}}-[f{PRODUCT_ID_CHARS}]{{3}}-[{PRODUCT_ID_CHARS}]{{3}}$"
```

Comment:
need to use raw strings for regex

```suggestion
    rf"^[{PRODUCT_ID_CHARS}]{{3}}-[f{PRODUCT_ID_CHARS}]{{3}}-[{PRODUCT_ID_CHARS}]{{3}}$"
```